pkg/pipelineascode/pipelineascode_test.go

```diff
@@ -130,6 +149,47 @@ func TestRun(t *testing.T) {
 			finalStatus:     "failure",
 			finalStatusText: "we need at least one pipelinerun to start with",
 		},
+		{
+			name: "pull request/unknown-remotetask-but-fail-on-matching",
+			runevent: info.Event{
+				SHA:           "principale",
+				Organization:  "organizationes",
+				Repository:    "lagaffe",
+				URL:           "https://service/documentation",
+				HeadBranch:    "press",
+				BaseBranch:    "main",
+				Sender:        "fantasio",
+				EventType:     "push",
+				TriggerTarget: "push",
+			},
+			tektondir:       "testdata/pull_request-nomatch-remotetask",
+			finalStatus:     "failure",
+			finalStatusText: "we need at least one pipelinerun to start with",
+		},
+		{
+			name: "pull request/match-but-fail-to-start-on-unknown-remotetask",
+			runevent: info.Event{
+				Event: &github.PullRequestEvent{
+					PullRequest: &github.PullRequest{
+						Number: github.Int(666),
+					},
+				},
+				SHA:               "fromwebhook",
+				Organization:      "owner",
+				Sender:            "owner",
+				Repository:        "repo",
+				URL:               "https://service/documentation",
+				HeadBranch:        "press",
+				BaseBranch:        "main",
+				EventType:         "pull_request",
+				TriggerTarget:     "pull_request",
+				PullRequestNumber: 666,
+				InstallationID:    1234,
+			},
+			tektondir:       "testdata/pull_request-nomatch-remotetask",
+			finalStatus:     "failure",
```

Comment:
can we have one test with success status

Reply:
i have added some failure tests, the success tests are already there (see coverage)
(this pr is mostly refactoring)